sql/parser/aggregate_builtins.go

```diff
@@ -204,120 +206,106 @@ func NewIdentAggregate() AggregateFunc {
 }
 
 // Add sets the value to the passed datum.
-func (a *identAggregate) Add(datum Datum) error {
+func (a *identAggregate) Add(datum Datum) {
 	a.val = datum
-	return nil
 }
 
 // Result returns the value most recently passed to Add.
-func (a *identAggregate) Result() (Datum, error) {
-	return a.val, nil
+func (a *identAggregate) Result() Datum {
+	return a.val
 }
 
 type avgAggregate struct {
-	sumAggregate
+	agg   AggregateFunc
```

Comment:
why change this to an interface type? ISTM that this will cause dynamic dispatch where static dispatch would otherwise be possible.